City-of-Helsinki/drupal-hdbt
# UHF-6157: Add support for toggling between local and global navigatio…

…n methods

# [UHF-6157](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6157)

As we want to move towards global hel.fi sitewide navigation from instance specific local navigation scheme, we need to add support for it. However, there will still be external instances where local (instance specific)navigation is preferred, so we need to support that too.

## What was done


* Current mobile navigation scss code was moved to local navigation
* Two new root scss files were created `nav-global.scss` and `nav-local.scss``
* These files were added as libraries to be included on page.
* Only one of the scss files gets included, based on if the hdbt global navigation module is enabled.
* Twig templates were modified to check if the module is enabled or not for further global nav development.

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-6157_local_and_global_nav`
* Run `make drush-cr`

## How to test


* [x] Check that the navigation by default (with the module uninstalled) still works as it did.
* [x] Enable the module
    * Either by clicking
        * Navigate to /admin/modules page (for example in [etusivu](https://helfi-etusivu.docker.so/fi/admin/modules) instance)
        * Search for `HDBT Global Navigation`
        * Enable it
    * Or by running this in shell 
        * Run `drush en hdbt_global_navigation`
* [x] Check that local navigation is no longer there, and there are ugly GLOBAL NAV texts instead as placeholder
* [x] Check that networks tab loads `nav_global.min.css` file instead of `nav_local.min.css` (and no `nav-local.min.js`)
* [x] Disable the module
    * Either by cliking
        * Navigate to /admin/modules/uninstall (for example in [etusivu](https://helfi-etusivu.docker.so/fi/admin/modules/uninstall)
        * Search for `HDBT Global Navigation`
        * Disable it, then confirm
    * Or by running this in shell
        * Run `drush pm-uninstall hdbt_global_navigation`
* [x] Check that navigation works again locally as it used to
* [x] Check that networks tab loads `nav_local.min.css` file instead of `nav_global.min.css` (and `nav-local.min.js`)

## Designers review
